### ISSUE TITLE
Some queries kill the Context Broker when dealing with Context Providers

### ISSUE THREAD

**Issue by dmoranj** [telefonicaid/fiware-orion#699]
The problem arises in the following scenario:
- An IoT Agent makes a context registration with the following data (this JSON reflects the object that is passed to the requests library, the actual payload is in the "json" attribute):

```
{
    "url": "http://192.168.56.101:1026/NGSI9/registerContext",
    "method": "POST",
    "json": {
        "contextRegistrations": [
            {
                "entities": [
                    {
                        "type": "Light",
                        "isPattern": "false",
                        "id": "TestClient:Light"
                    }
                ],
                "attributes": [
                    {
                        "name": "luminescence",
                        "type": "Lumens",
                        "isDomain": "false"
                    }
                ],
                "providingApplication": "http://192.168.56.1:4041"
            }
        ],
        "duration": "P1M"
    },
    "headers": {
        "fiware-service": "smartGondor",
        "fiware-servicepath": "/gardens"
    }
}
```
- Once the context is registered, the following queryContext request is issued to the Context Broker:
```
{
    "url": "http://192.168.56.101:1026/v1/queryContext",
    "method": "POST",
    "headers": {
        "fiware-service": "smartGondor",
        "fiware-servicepath": "/gardens"
    },
    "json": {
        "entities": [
            {
                "type": "Light",
                "id": "TestClient:Light",
                "isPattern": "false"
            }
        ],
        "attributes": []
    }
}
```
The answer to the last query is an exception: "Error: socket hang up". This exception arises when the Context Broker dies without sending a response. 

No log entry is written to the Context  Broker log. `dmesg` shows the following entries, one for each crash:
```
libmicrohttpd[3894]: segfault at 0 ip 00000000005923a7 sp 00007f5563fe4dc0 error 4 in contextBroker[400000+313000]
libmicrohttpd[4002]: segfault at 0 ip 00000000005923a7 sp 00007f2d72be2dc0 error 4 in contextBroker[400000+313000]
libmicrohttpd[4174]: segfault at 0 ip 00000000005923a7 sp 00007f6acb5e3dc0 error 4 in contextBroker[400000+313
 ```
The `messages` log file shows the following lines for the error:
```
Dec 30 10:52:21 CentollOS kernel: libmicrohttpd[4359]: segfault at 0 ip 00000000005923a7 sp 00007f29a3fe4dc0 error 4 in contextBroker[400000+313000]
Dec 30 10:52:21 CentollOS abrtd: Directory 'ccpp-2014-12-30-10:52:21-4323' creation detected
Dec 30 10:52:21 CentollOS abrt[4360]: Saved core dump of pid 4323 (/usr/bin/contextBroker) to /var/spool/abrt/ccpp-2014-12-30-10:52:21-4323 (178573312 bytes)
Dec 30 10:52:21 CentollOS abrtd: Package 'contextBroker' isn't signed with proper key
Dec 30 10:52:21 CentollOS abrtd: 'post-create' on '/var/spool/abrt/ccpp-2014-12-30-10:52:21-4323' exited with 1
Dec 30 10:52:21 CentollOS abrtd: Deleting problem directory '/var/spool/abrt/ccpp-2014-12-30-10:52:21-4323'
```

**Comment by fgalan** [comment]
Fixed in PR #700, but waiting some testing from @dmoranj before clossing definitively the issue.

**Comment by dmoranj** [comment]
Fixed without problems.
